python/mspasspy/db/database.py

```diff
@@ -570,6 +571,150 @@ def update_ensemble_metadata(self, ensemble_object, include_undefined=False, exc
     # TODO: the following is not used when data is read from the database. We need
     #       link these metadata keys with the actual member variables just like the
     #       npts or t0 in TimeSeries.
+    def _load_collection_metadata(self,mspass_object, exclude_keys, include_undefined=False, collection=None):
+        """
+        Master Private Method
+
+        Reads metadata from a requested collection and loads standard attributes from collection to the data passed as mspass_object.
+        The method will only work if mspass_object has the collection_id attribute set to link it to a unique document in source.  
+
+        :param mspass_object:   data where the metadata is to be loaded
+        :type mspass_object:  must be TimeSeries, Seismogram, TimeSeriesEnsemble, or SeismogramEnsemble.
+        
+        :param exclude_keys: list of attributes that should not normally be loaded.
+        Default attributes not normally need that are loaded from stationxml.  Ignored if include_undefined is set True.
+        
+        :param include_undefined:  when true all data in the matching document are loaded.
+
+        :param collection: requested collection metadata should be loaded
+        :type collection: str 
+
+        :exception:  any detected errors will cause a MsPASSError to be thrown
+        (colleagues:  this may be wrong sphynx syntax for defining an exception)
+        """
+        if collection not in ["source", "site", "channel"]:
```

Comment:
This is not necessary as this method should be flexible for any collections. It will check whether `collection + '_id' exists later so, if a collection is not defined in the object, it will throw an error.
`